### DISCUSSION TITLE
Container Trust Revamp?

### DISCUSSION THREAD

**Discussion by Blzd87** [GriefPrevention/GriefPrevention#1720] [Ideas]
would be great if this permission only allowed players to interact with animals like feed them and milk etc but NOT kill them and also villagers only allow trading with them but not killing them..

some of my players have been nice to make beef farms and small villager trading setups with container trust set to public so newer players can get basic stuff from the community area but we get the bad players who just flat out kill the cows and villagers when they realise they can... which is bad.

or can there be a new perm Interact Trust? so it stops them from being killed and still allows players to open chests under cow farm to get the beef and leather?

just an idea hope you consider it thanks.

Regards

Blzd
MC Gods SMP Owner

**Comment by Jikoo** [comment]
Killing villagers requires full trust, so that shouldn't be a concern. Most of your other issues could probably be resolved by some careful (sub)claiming.

If you want to create a claim that generally everyone can access fully, like a claim over a public village, that's fine. You could `/containertrust public` in it. If you have animal pens inside the village you want to protect, you could `/restrictsubclaim` the subclaim over the pen so that not everyone can kill them. Then you could have publicly accessible chests out in front of the pen while still protecting the animals from grief.
/e Should also note that automatic chicken farms are probably the simplest way to keep a pure vanilla infinite meat supply on hand, and while a hacked client could kill the chickens inside if you're not running an anticheat, they're generally sealed away from attacks otherwise.

Breeding and killing at the same level generally makes sense, particularly for things like cows unfortunately. I agree for sheep it makes sense not to be the case for additional interactions like shearing, but consistency is also important for player understanding. If that's how cows and pigs work, it should also be how sheep work even if it's not completely optimal. An addon can change that later, but base GP is trying to apply simple, consistent, broad rules everyone can grasp easily. If you can breed but not kill you'd still be free to wildly overpopulate pens.
